badge_server/main.py

```diff
@@ -280,22 +286,30 @@ def _get_badge_status(
     The badge status will determine the right hand text and the color of
     the badge.
 
-    See badge_utils.STATUS_COLOR_MAPPING.
-    """
-    dep_status = dependency_res['status']
-    dep_status = 'SUCCESS' if dep_status == 'UP_TO_DATE' else dep_status
+    Args:
+        self_compat_res: a dict containing a package's self compatibility
+            status for py2 and py3. See _get_self_compatibility_dict().
+        google_compat_res: a dict containing a package's pair compatibility
+            status for py2 and py3. See _get_pair_compatibility_dict().
+        dependency_res: a dict containing a package's dependency status
+            for py2 and py3. See _get_dependency_dict().
 
+    Returns:
+        The cumulative badge status.
+    """
     statuses = []
     for pyver in ['py2', 'py3']:
         statuses.append(self_compat_res[pyver]['status'])
         statuses.append(google_compat_res[pyver]['status'])
-    statuses.append(dep_status)
+    statuses.append(dependency_res['status'])
+
+    if all(status == BadgeStatus.SUCCESS for status in statuses):
+        return BadgeStatus.SUCCESS
 
-    if all(status == 'SUCCESS' for status in statuses):
-        return 'SUCCESS'
-    elif 'UNKNOWN' in statuses:
-        return 'UNKNOWN'
-    return 'CHECK_WARNING'
+    # This works because the enums are stored in an ordered dict.
+    for status in BadgeStatus.__members__.values():
```

Comment:
Maybe make a new function: `get_highest_priority_status`?

Reply:
Can I actually do this in a follow up PR?

Reply:
Sure.